cloudprovider/alibabacloud/slb.go

```diff
@@ -144,12 +144,16 @@ func initLbCache(svcList []corev1.Service, minPort, maxPort int32, blockPorts []
 			var ports []int32
 			for _, port := range getPorts(svc.Spec.Ports) {
 				if port <= maxPort && port >= minPort {
-					newCache[lbId][port] = true
-					ports = append(ports, port)
+					value, ok := newCache[lbId][port]
+					if !ok || !value {
+						newCache[lbId][port] = true
+						ports = append(ports, port)
+					}
 				}
 			}
 			if len(ports) != 0 {
 				newPodAllocate[svc.GetNamespace()+"/"+svc.GetName()] = lbId + ":" + util.Int32SliceToString(ports, ",")
+				log.Infof("svc %s/%s allocate slb %s ports %v", svc.Namespace, svc.Name, lbId, ports)
```

Comment:
The log is unnecessary, cause podAllocate will be printed when finished the initialization.

Reply:
yes ,,,